### DISCUSSION TITLE
burn: Use same file for two ExePackage elements (Upgrade v3 to v4)

### DISCUSSION THREAD

**Discussion by mistoll** [wixtoolset/issues#7431] [Questions] [answered]
In v3 I used the following snippet to install SQL Express:
```
<PackageGroup Id="SqlExpress">
  <ExePackage Id="SqlExpress" DisplayName="SQL Server 2019 Express"
    Cache="keep" Compressed="no" PerMachine="yes" Permanent="yes" Vital="yes"
    Name="SQLEXPR_x64_ENU.exe"
    SourceFile="SQLEXPR_x64_ENU.exe"
    DownloadUrl="$(var.SqlWebLink)"
    DetectCondition="SqlInstanceFound"
    InstallCondition="INSTALL_PREREQUISITES=1"
    InstallArguments="$(var.SqlPackParameters) /ACTION=Install $(var.SqlInstanceParameters) /Q /HIDECONSOLE /SkipRules=RebootRequiredCheck"
    UninstallArguments="$(var.SqlPackParameters) /Action=Uninstall $(var.SqlFeatureParameters) /Q /HIDECONSOLE">
    <ExitCode Value="3010" Behavior="forceReboot" />
  </ExePackage>
 <ExePackage Id="SqlExpressUpgrade" DisplayName="SQL Server 2019 Express Upgrade"
    Cache="remove" Compressed="no" PerMachine="yes" Permanent="yes" Vital="yes"
    Name="SQLEXPR_x64_ENU.exe"
    SourceFile="$(var.DownloadsDir)SQLEXPR_x64_ENU.exe"
    DownloadUrl="$(var.SqlWebLink)"
    DetectCondition="NOT (SqlInstanceFound AND (SqlVersion &lt; v15.0.0.0))"
    InstallArguments="$(var.SqlPackParameters) /ACTION=Upgrade $(var.SqlFeatureParameters) /Q /HIDECONSOLE /SkipRules=RebootRequiredCheck">
    <ExitCode Value="3010" Behavior="forceReboot" />
  </ExePackage>
</PackageGroup>
```

The Idea behind that is to have different arguments on (first) installation and on upgrade.

In v4 this produces the error
`The CacheId '9774C72F4A0D16FA4473138E7F02CEEA62908C6D5E5D757EBE4C09ABA7999B7C' for package 'SqlExpressUpgrade' is duplicated. Each package must have a unique CacheId.`
If I add a CacheId Attribute the error becomes:
`The external Payload 'SqlExpressUpgrade' has a duplicate Name 'SQLEXPR_x64_ENU.exe'. When building the bundle or laying out the bundle, the file will get overwritten.`

If I change the name the file would be downloaded and cached twice, which is not what I want. Is it possible to use the same file for two ExePackages in v4? How?
If not, are there other options to switch the InstallArguments?

**Comment by rseanhall** [comment, marked as answer]
No, you can't use the same file for two ExePackages in v4 because it doesn't work properly at runtime.

The [`CommandLine`](https://wixtoolset.org/docs/schema/wxs/commandline/) element can be used to provide conditional arguments to an ExePackage.
